kalite/topic_tools/annotate.py

```diff
@@ -50,97 +50,71 @@ def update_content_availability(content_list, language="en"):
-                        items.append(item)
-                        update["available"] = True
-                update["all_assessment_items"] = items
-            else:
-                exercise_file = content.get("name", "") + ".html"
-
-                if language == "en":
-                    exercise_template = exercise_file
-                    if exercise_template in exercise_templates:
-                        update["available"] = True
-                        update["template"] = exercise_template
-                else:
-                    exercise_template = os.path.join(language, exercise_file)
-                    if exercise_template in exercise_templates:
-                        update["available"] = True
-                        update["template"] = exercise_template
+            # Databases have been pre-filtered to only contain existing exercises
+            if database_exists(language=language, channel=channel):
+                update["available"] = True
 
         elif content.get("kind") == "Topic":
             # Ignore topics, as we only want to update their availability after we have updated the rest.
             continue
         else:
+            file_id = content.get("youtube_id", content.get("id"))
             default_thumbnail = create_thumbnail_url(content.get("id"))
-            dubmap = i18n.get_id2oklang_map(content.get("id"))
-            if dubmap:
-                content_lang = i18n.select_best_available_language(language, available_codes=dubmap.keys()) or ""
-                if content_lang:
-                    dubbed_id = dubmap.get(content_lang)
-                    format = content.get("format", "")
-                    if (dubbed_id + "." + format) in contents_folder:
-                        update["available"] = True
-                        thumbnail = create_thumbnail_url(dubbed_id) or default_thumbnail
-                        update["content_urls"] = {
-                            "stream": django_settings.CONTENT_URL + dubmap.get(content_lang) + "." + format,
-                            "stream_type": "{kind}/{format}".format(kind=content.get("kind").lower(), format=format),
-                            "thumbnail": thumbnail,
-                        }
-                    elif django_settings.BACKUP_VIDEO_SOURCE:
-                        update["available"] = True
-                        update["content_urls"] = {
-                            "stream": django_settings.BACKUP_VIDEO_SOURCE.format(youtube_id=dubbed_id, video_format=format),
-                            "stream_type": "{kind}/{format}".format(kind=content.get("kind").lower(), format=format),
-                            "thumbnail": django_settings.BACKUP_VIDEO_SOURCE.format(youtube_id=dubbed_id, video_format="png"),
-                        }
+            format = content.get("format", "")
+            filename = file_id + "." + format
```

Comment:
Not hugely intentional - if something doesn't have a format, we're probably not going to find the file.